api/admin/controller.py

```diff
@@ -2323,6 +2325,109 @@ def _sitewide_settings_controller(self, configuration_object):
+                         SharedODLAPI,
+                         FeedbooksOPDSImporter,
+                        ]
+
+        self_test_results = None
+
+        if not collection or not collection.protocol:
+            return None
+
+        if collection.protocol == OPDSImportMonitor.PROTOCOL:
+            newCollection["protocolClass"] = OPDSImportMonitor
+
+        if newCollection["protocolClass"] in provider_apis and issubclass(newCollection["protocolClass"], HasSelfTests):
+            if (collection.protocol == OPDSImportMonitor.PROTOCOL):
+                self_test_results = newCollection["protocolClass"].prior_test_results(self._db, newCollection["protocolClass"], self._db, collection, OPDSImporter)
+            else:
+                self_test_results = newCollection["protocolClass"].prior_test_results(self._db, newCollection["protocolClass"], self._db, collection)
+
+        return self_test_results
+
+    def collection_self_tests(self, identifier):
+        provider_apis = [OPDSImporter,
+                         OPDSForDistributorsAPI,
+                         OverdriveAPI,
+                         OdiloAPI,
+                         BibliothecaAPI,
+                         Axis360API,
+                         OneClickAPI,
+                         EnkiAPI,
+                         ODLWithConsolidatedCopiesAPI,
+                         SharedODLAPI,
+                         FeedbooksOPDSImporter,
+                        ]
+        protocols = self._get_protocols(provider_apis)
+
+        if not identifier:
+            return Response("No Identifier", 200)
+
+        collection = dict()
+        collectionFound = None
+        for col in self._db.query(Collection).order_by(Collection.name).all():
+            if col.id == int(identifier):
+                collectionFound = col
+                collection = dict(
+                    id=col.id,
+                    name=col.name,
+                    protocol=col.protocol,
+                    parent_id=col.parent_id,
+                    settings=dict(external_account_id=col.external_account_id),
+                    protocolClass=None,
+                )
+
+                if col.protocol in [p.get("name") for p in protocols]:
+                    [protocol] = [p for p in protocols if p.get("name") == col.protocol]
+
+                    [protocolClass] = [p for p in provider_apis if p.NAME == col.protocol]
+                    collection["protocolClass"] = protocolClass
+
+                collection["self_test_results"] = self._get_prior_test_results(col, collection)
+
+        if flask.request.method == 'GET':
+            if "protocolClass" in collection:
+                del collection["protocolClass"]
+            return dict(collection=collection)
+
+        if flask.request.method == "POST":
+            if collection["protocolClass"]:
+                if (collection["protocol"] == OPDSImportMonitor.PROTOCOL):
```

Comment:
I think there's this and possibly Feedbooks that are special cases. The `OPDSImporter` class doesn't inherit from `HasSelfTests` so that's why I'm making this case here.